clio-client/src/main/scala/org/broadinstitute/clio/client/commands/MoveWgsUbamCommand.scala

```diff
@@ -31,63 +33,97 @@ object MoveWgsUbamCommand
+      _ <- verifyPath(webClient, config) withErrorMsg
+        "Clio client can only handle cloud operations right now."
+      wgsUbamPath <- queryForWgsUbamPath(webClient, config) withErrorMsg
+        "Could not query the WgsUbam. No files have been moved."
+      _ <- copyGoogleObject(wgsUbamPath, config.ubamPath, ioUtil) withErrorMsg
+        "An error occurred while copying the files in the cloud. No files have been moved."
       upsertUbam <- upsertUpdatedWgsUbam(webClient, config) withErrorMsg
-        """An error occurred while upserting the WgsUbam. The state of clio is now inconsistent.
+        s"""An error occurred while upserting the WgsUbam.
           |The ubam exists in both at both the old and the new locations.
           |At this time, Clio only knows about the bam at the old location.
-          |Then, try upserting (not moving) the unmapped bam with the correct path then deleting the old bam.
-          |If the state of Clio cannot be fixed, please contact the Green Team at greenteam@broadinstitute.org
+          |Try removing the ubam at the new location and re-running this command.
+          |If this cannot be done, please contact the Green Team at ${ClioClientConfig.greenTeamEmail}.
         """.stripMargin
-      _ <- deleteGoogleObject(wgsUbam.ubamPath, ioUtil) withErrorMsg
-        """The old bam was not able to be deleted. Clio has been updated to point to the new bam.
-          | Please delete the old bam. If this cannot be done, contact Green Team at greenteam@broadinstitute.org
+      _ <- deleteGoogleObject(wgsUbamPath, ioUtil) withErrorMsg
+        s"""The old bam was not able to be deleted. Clio has been updated to point to the new bam.
+          | Please delete the old bam. If this cannot be done, contact Green Team at ${ClioClientConfig.greenTeamEmail}.
           | """.stripMargin
     } yield {
+      logger.info(
+        s"Successfully moved '${wgsUbamPath.get}' to '${config.ubamPath.get}'"
+      )
       upsertUbam
     }
   }
 
-  private def queryForWgsUbam(
-    webClient: ClioWebClient,
-    config: BaseArgs
-  ): Future[TransferWgsUbamV1QueryOutput] = {
-    val httpResponse = webClient.queryWgsUbam(
-      config.bearerToken.getOrElse(""),
-      TransferWgsUbamV1QueryInput(
-        flowcellBarcode = config.flowcell,
-        lane = config.lane,
-        libraryName = config.libraryName,
-        location = Option(Location.GCP),
-        documentStatus = Option(DocumentStatus.Normal)
-      )
-    )
+  private def verifyPath(webClient: ClioWebClient,
+                         config: BaseArgs): Future[Unit] = {
+    implicit val executionContext: ExecutionContext = webClient.executionContext
+    Future(config.location.foreach {
+      case "GCP" => ()
+      case _ =>
+        throw new Exception(
+          "Only GCP unmapped bams are supported at this time."
+        )
+    }).flatMap { _ =>
+      Future(config.ubamPath.foreach {
```

Comment:
This doesn't need to be a `flatMap`, the `Future` body isn't actually doing anything async. You could either change the `flatMap` to a `map` or just move the `foreach` up into the Future above, under the other `foreach`.